File: desktopowe2tiGim/Obiektowe_GIM1/Obiektowe_GIM1/Person.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Obiektowe_GIM1
{
    public class Person
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime DateOfBirth { get; set; }

        public Person() {
            Console.WriteLine($"Wywolanie konstruktora domyslnego: {GetType()}");
            FirstName = "Adam";
            LastName = "Nowak";
            DateOfBirth = new DateTime(2000,1,1);
        }

        public Person(string firstName, string lastName, DateTime dateOfBirth) {
            FirstName = firstName;
            LastName = lastName;
            DateOfBirth = dateOfBirth;
        }
        public  string Info() {
            StringBuilder sb = new StringBuilder();
            sb.Append("\n======== Informacje o osobie ============").Append(Environment.NewLine);
            sb.Append($"Imię: {FirstName} nazwisko: {LastName} data urodzenia: {DateOfBirth.Date.ToShortDateString()}")
                .Append(Environment.NewLine);
            return sb.ToString();
        }
    }
}

File: zdalnecppGdow/klasy_dziedzczenie1/src/Person.cpp
#include "Person.h"

Person::Person()
{
   firstName = "Adam";
   lastName = "Nowak";
   age = 20;
}
Person::Person(string firstName,string lastName,int age){
    this->firstName = firstName;
    this->lastName = lastName;
    SetAge(age);
}
void Person::Info(){
    cout << " --------------- Informacja o osobie --------------------\n";
    cout << "      Imie: " << firstName<<endl;
    cout << "  Nazwisko: " << lastName<<endl;
    cout << "      Wiek: " << age<<endl;

}
ostream& operator<<(ostream& os, const Person& p){
      os << " --------------- Informacja o osobie --------------------\n";
    os << "      Imie: " << p.firstName<<endl;
    os << "  Nazwisko: " << p.lastName<<endl;
    os << "      Wiek: " << p.age<<endl;
    return os;
}

Person::~Person()
{
    //dtor
}
